ThePalaceProject/web-opds-client
# Upgrade to Node 18.

## Description

Upgrades this project to build with Node 18, along with some other quality/security improvements. This includes:

- Upgrading dependencies that were incompatible with Node 18
- Upgrading dependencies to resolve `npm audit` security errors (not all of them, just the easy ones for now)
- Code/configuration changes required by major version upgrades of dependencies, including
  - Major typescript update required adding some missing type parameters to Promise constructors
  - Major webpack update required updating some webpack config files, moving some npm script arguments into webpack config files, and removing some no-longer-supported arguments
- Some minor formatting changes to resolve lint errors
- Additional `export`s from some files to resolve documentation generation warnings

## Motivation and Context

The project had been using Node 12, which is no longer supported, and will not be allowed in GitHub actions soon. Notion: https://www.notion.so/lyrasis/Upgrade-web-opds-client-to-Node-18-44f9a03cbb0545cc863f19a5db8bb9c0

## How Has This Been Tested?

- Verify that `npm install` runs without error
- Verify that all npm scripts in all package.json files (package.json, packages/server/package.json, and packages/web-opds-client/package.json) run without error (except for `npm publish`, which we'll test when we actually publish). This especially includes `npm test` in packages/web-opds-client.
- Verify that the app runs in dev:
  - `npm run dev`
  - Open http://localhost:3000 in a browser
  - Enter a feed URL, e.g. https://cerberus.tpp-qa.lyrasistechnology.org/cerberus-test-1/feed/1206
  - Verify the feed is displayed

## Checklist:
